// @vueuse/integrations/useChangeCase.iife.js
;var VueDemi = (function (VueDemi, Vue, VueCompositionAPI) {
  if (VueDemi.install) {
    return VueDemi
  }
  if (Vue) {
    if (Vue.version.slice(0, 4) === '2.7.') {
      for (var key in Vue) {
        VueDemi[key] = Vue[key]
      }
      VueDemi.isVue2 = true
      VueDemi.isVue3 = false
      VueDemi.install = function (){}
      VueDemi.Vue = Vue
      VueDemi.Vue2 = Vue
      VueDemi.version = Vue.version
    }
    else if (Vue.version.slice(0, 2) === '2.') {
      if (VueCompositionAPI) {
        for (var key in VueCompositionAPI) {
          VueDemi[key] = VueCompositionAPI[key]
        }
        VueDemi.isVue2 = true
        VueDemi.isVue3 = false
        VueDemi.install = function (){}
        VueDemi.Vue = Vue
        VueDemi.Vue2 = Vue
        VueDemi.version = Vue.version
      } else {
        console.error(
          '[vue-demi] no VueCompositionAPI instance found, please be sure to import `@vue/composition-api` before `vue-demi`.'
        )
      }
    } else if (Vue.version.slice(0, 2) === '3.') {
      for (var key in Vue) {
        VueDemi[key] = Vue[key]
      }
      VueDemi.isVue2 = false
      VueDemi.isVue3 = true
      VueDemi.install = function (){}
      VueDemi.Vue = Vue
      VueDemi.Vue2 = undefined
      VueDemi.version = Vue.version
      VueDemi.set = function(target, key, val) {
        if (Array.isArray(target)) {
          target.length = Math.max(target.length, key)
          target.splice(key, 1, val)
          return val
        }
        target[key] = val
        return val
      }
      VueDemi.del = function(target, key) {
        if (Array.isArray(target)) {
          target.splice(key, 1)
          return
        }
        delete target[key]
      }
    } else {
      console.error('[vue-demi] Vue version ' + Vue.version + ' is unsupported.')
    }
  } else {
    console.error(
      '[vue-demi] no Vue instance found, please be sure to import `vue` before `vue-demi`.'
    )
  }
  return VueDemi
})(
  this.VueDemi = this.VueDemi || (typeof VueDemi !== "undefined" ? VueDemi : {}),
  this.Vue || (typeof Vue !== "undefined" ? Vue : undefined),
  this.VueCompositionAPI || (typeof VueCompositionAPI !== "undefined" ? VueCompositionAPI : undefined)
);
;
;(function (exports, shared, vueDemi, changeCase$1) {
  'use strict';

  var changeCase = /*#__PURE__*/Object.freeze({
    __proto__: null,
    camelCase: changeCase$1.camelCase,
    capitalCase: changeCase$1.capitalCase,
    constantCase: changeCase$1.constantCase,
    dotCase: changeCase$1.dotCase,
    headerCase: changeCase$1.headerCase,
    noCase: changeCase$1.noCase,
    paramCase: changeCase$1.paramCase,
    pascalCase: changeCase$1.pascalCase,
    pathCase: changeCase$1.pathCase,
    sentenceCase: changeCase$1.sentenceCase,
    snakeCase: changeCase$1.snakeCase
  });

  function useChangeCase(input, type, options) {
    if (shared.isFunction(input))
      return vueDemi.computed(() => changeCase[type](shared.resolveUnref(input), options));
    const text = vueDemi.ref(input);
    return vueDemi.computed({
      get() {
        return changeCase[type](text.value, options);
      },
      set(value) {
        text.value = value;
      }
    });
  }

  exports.useChangeCase = useChangeCase;

  Object.defineProperty(exports, '__esModule', { value: true });

})(this.VueUse = this.VueUse || {}, VueUse, VueDemi, changeCase);
